Rancher IPv6 Support (#2041)
rancher/rancher-docs: docs/reference-guides/dual-stack.md

---
title: IPv4/IPv6 Dual-stack
---

<head>
  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/reference-guides/dual-stack/"/>
</head>

Kubernetes supports IPv4-only, IPv6-only, and dual-stack networking configurations.
For more details, refer to the official [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).

## Installing Rancher on IPv6-Only or Dual-Stack Clusters

Rancher can run on clusters using:

- IPv4-only
- IPv6-only
- Dual-stack (IPv4 + IPv6)

> When Rancher is installed on an **IPv6-only cluster**, it can communicate externally **only over IPv6**. This means it can provision:

Reviewer: ```suggestion
When you install Rancher on an **IPv6-only cluster**, it can communicate externally **only over IPv6**. This means it can provision:
```

- IPv6-only clusters
- Dual-stack clusters  
  _(IPv4-only downstream clusters are not possible in this case)_

When Rancher is installed on a **dual-stack cluster**, it can communicate over both IPv4 and IPv6, and can therefore provision:

- IPv4-only clusters
- IPv6-only clusters
- Dual-stack clusters

For installation steps, see the guide: **[Installing and Upgrading Rancher](../getting-started/installation-and-upgrade/installation-and-upgrade.md)**.

#### Requirement for the Rancher Server URL

When provisioning IPv6-only downstream clusters, the **Rancher Server URL must be reachable over IPv6**, because downstream nodes connect back to the Rancher server using IPv6.

## Provisioning IPv6-Only or Dual-Stack Clusters

> Starting with **Rancher v2.13.0**, you can provision RKE2 and K3s **Node driver** (machine pools) or **Custom cluster** (existing hosts) clusters using IPv4-only, IPv6-only, or dual-stack networking.

Reviewer: ```suggestion
You can provision RKE2 and K3s **Node driver** (machine pools) or **Custom cluster** (existing hosts) clusters using IPv4-only, IPv6-only, or dual-stack networking.
```

For more information, see the [Kubernetes Documentation](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#windows-support).

K3s does **not** support Windows ([FAQ](https://docs.k3s.io/faq#does-k3s-support-windows))

RKE2 supports Windows, but requires using either `Calico` or `Flannel` as the CNI.  
Note that dual-stack clusters using BGP are not currently supported on Windows installations of RKE2.  
For more details, see [RKE2 Network Options](https://docs.rke2.io/networking/basic_network_options).

### Provisioning Node Driver Clusters

Rancher currently supports assigning IPv6 addresses in **node driver** clusters with:

- [Amazon EC2](../how-to-guides/new-user-guides/launch-kubernetes-with-rancher/use-new-nodes-in-an-infra-provider/create-an-amazon-ec2-cluster.md)
- [DigitalOcean](../how-to-guides/new-user-guides/launch-kubernetes-with-rancher/use-new-nodes-in-an-infra-provider/create-a-digitalocean-cluster.md)

Support for additional providers will be introduced in future releases.

:::note DigitalOcean Limitation

Creating an **IPv6-only cluster** using the DigitalOcean node driver is currently **not supported**.

> For more details, please see [rancher/rancher#52523](https://github.com/rancher/rancher/issues/52523#issuecomment-3457803572)

Reviewer: ```suggestion
For more details, please see [rancher/rancher#52523](https://github.com/rancher/rancher/issues/52523#issuecomment-3457803572).
```
